src/content_mapblock.cpp

```diff
@@ -373,12 +379,46 @@ void MapblockMeshGenerator::drawAutoLightedCuboid(aabb3f box, const f32 *txc,
 			d.Z = (j & 1) ? dz2 : dz1;
 			lights[j] = blendLight(d);
 		}
-		drawCuboid(box, tiles, tile_count, lights, txc);
+		drawCuboid(box, tiles, tile_count, lights, txc, faces);
 	} else {
-		drawCuboid(box, tiles, tile_count, nullptr, txc);
+		drawCuboid(box, tiles, tile_count, nullptr, txc, faces);
 	}
 }
 
+void MapblockMeshGenerator::drawSolidNode()
+{
+	bool faces[6] = {0, 0, 0, 0, 0, 0};
+	static const v3s16 tile_dirs[6] = {
```

Comment:
no need for this you already have this in directiontables.h if i'm correct

Reply:
sadly no, the order is different, and it is important here.